astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,427 @@
+                altitude = u.Quantity(location[2]).to("m").value
+
+        if int(neptune_arcmodel) not in [1, 2, 3]:
+            raise ValueError(
+                f"Illegal Neptune arc model {neptune_arcmodel}. must be one of 1, 2, or 3 (see https://pds-rings.seti.org/tools/viewer3_nep.shtml for details)"
+            )
+
+        # configure request_payload for ephemeris query
+        request_payload = dict(
+            [
+                ("abbrev", planet[:3]),
+                ("ephem", planet_defaults[planet]["ephem"]),
+                ("time", epoch.utc.iso[:16]),
+                ("fov", 10),  # figure option - hardcode ok
+                ("fov_unit", planet.capitalize() + " radii"),
+                ("center", "body"),
+                ("center_body", planet.capitalize()),
+                ("center_ansa", planet_defaults[planet]["center_ansa"]),
+                ("center_ew", "east"),
+                ("center_ra", ""),
+                ("center_ra_type", "hours"),
+                ("center_dec", ""),
+                ("center_star", ""),
+                ("viewpoint", viewpoint),
+                ("observatory", observatory),
+                ("latitude", latitude),
+                ("longitude", longitude),
+                ("lon_dir", "east"),
+                ("altitude", altitude),
+                ("moons", planet_defaults[planet]["moons"]),
+                ("rings", planet_defaults[planet]["rings"]),
+                ("arcmodel", neptune_arcmodels[int(neptune_arcmodel)]),
+                ("extra_ra", ""),  # figure options below this line, can all be hardcoded and ignored
+                ("extra_ra_type", "hours"),
+                ("extra_dec", ""),
+                ("extra_name", ""),
+                ("title", ""),
+                ("labels", "Small (6 points)"),
+                ("moonpts", "0"),
+                ("blank", "No"),
+                ("opacity", "Transparent"),
+                ("peris", "None"),
+                ("peripts", "4"),
+                ("arcpts", "4"),
+                ("meridians", "Yes"),
+                ("output", "html"),
+            ]
+        )
+
+        # return request_payload if desired
+        if get_query_payload:
+            return request_payload
+
+        # query and parse
+        response = self._request(
+            "GET", self._url, params=request_payload, timeout=self.timeout, cache=cache
+        )
+
+        return response
+
+    def _parse_result(self, response, verbose=None):
+        """
+        Routine for parsing data from ring node
+
+        Parameters
+        ----------
+        self : RingNodeClass instance
+        response : list
+            raw response from server
+
+
+        Returns
+        -------
+        bodytable : `astropy.QTable`
+        ringtable : `astropy.QTable`
+        """
+        try:
+            self._last_query.remove_cache_file(self.cache_location)
+        except (FileNotFoundError, OSError):
```

Comment:
`FileNotFoundError` is a subclass of `OSError`, so I don't see why it would be necessary to check for both.